code/datums/mutations/holy_mutation/honorbound.dm

```diff
@@ -0,0 +1,125 @@
+	locked = TRUE
+	text_gain_indication = "<span class='notice'>You feel honorbound!</span>"
+	text_lose_indication = "<span class='warning'>You feel unshackled from your code of honor!</span>"
+	var/list/guilty = list() //list of guilty people
+
+/datum/mutation/human/honorbound/on_acquiring(mob/living/carbon/human/owner)
+	if(..())
+		return
+
+	//moodlet
+	SEND_SIGNAL(owner, COMSIG_ADD_MOOD_EVENT, "honorbound", /datum/mood_event/honorbound)
+	//signals that check for guilt
+	RegisterSignal(owner, COMSIG_PARENT_ATTACKBY, .proc/attackby_guilt)
+	RegisterSignal(owner, COMSIG_ATOM_HULK_ATTACK, .proc/hulk_guilt)
+	RegisterSignal(owner, COMSIG_ATOM_ATTACK_HAND, .proc/hand_guilt)
+	RegisterSignal(owner, COMSIG_ATOM_ATTACK_PAW, .proc/paw_guilt)
+	RegisterSignal(owner, COMSIG_ATOM_BULLET_ACT, .proc/bullet_guilt)
+	RegisterSignal(owner, COMSIG_ATOM_HITBY, .proc/thrown_guilt)
+
+	//signal that checks for dishonorable attacks
+	RegisterSignal(owner, COMSIG_MOB_CLICKON, .proc/attack_honor)
+
+/datum/mutation/human/honorbound/on_losing(mob/living/carbon/human/owner)
+	UnregisterSignal(owner, list(COMSIG_PARENT_ATTACKBY,COMSIG_ATOM_HULK_ATTACK,COMSIG_ATOM_ATTACK_HAND,COMSIG_ATOM_ATTACK_PAW,COMSIG_ATOM_BULLET_ACT,COMSIG_ATOM_HITBY,COMSIG_MOB_CLICKON))
+	. = ..()
+
+/datum/mutation/human/honorbound/proc/attack_honor(mob/living/carbon/human/honorbound, atom/clickingon, params)
+	SIGNAL_HANDLER
+	var/obj/item/weapon = honorbound.get_active_held_item()
+	var/list/modifiers = params2list(params)
+
+	if(!isliving(clickingon))
+		return
+	//fucking YUGE check but essentially is every situation where a click shouldn't be checked for honor:
+	//non combat mode clicks:
+	//* no weapon, not disarming, weapon has no force
+	//combat mode clicks
+	//* is not a gun and out of range of attacking
+	if(!honorbound.DirectAccess(clickingon) && !isgun(weapon))
+		return
+	if(!honorbound.combat_mode && ((!weapon || !weapon.force) && !LAZYACCESS(modifiers, RIGHT_CLICK)))
+		return
+	var/mob/living/clickedmob = clickingon
+	if(!is_honorable(honorbound, clickedmob))
+		return COMSIG_MOB_CANCEL_CLICKON
+
+/datum/mutation/human/honorbound/proc/guilty(mob/living/user)
+	if(user in guilty)
+		return
+	var/datum/mind/guilty_conscience = user.mind
+	if(guilty_conscience) //sec and medical are immune to becoming guilty through attack (we don't check holy because holy shouldn't be able to attack eachother anyways)
+		var/job = guilty_conscience.assigned_role
+		if(job in (GLOB.security_positions + GLOB.medical_positions))
+			return
+	to_chat(owner, "<span class='notice'>[user] is now considered guilty by [GLOB.deity] for attacking you first.</span>")
+	guilty += user
+
+/datum/mutation/human/honorbound/proc/is_honorable(mob/living/carbon/human/honorbound_human, mob/living/target_creature)
+	var/is_guilty = (target_creature in guilty)
+	//THE UNREADY (Applies over ANYTHING else!)
+	if(target_creature.IsSleeping() || HAS_TRAIT(target_creature, TRAIT_RESTRAINED))
```

Comment:
should this also check for unconsciousness, which I believe is different from sleeping?